src/analyses/natural_loops.h

```diff
@@ -20,6 +30,12 @@ Author: Georg Weissenbacher, georg@weissenbacher.name
 
 #include "cfg_dominators.h"
 
+/// Main driver for working out if a class (normally goto_programt) has any natural loops.
+/// \ref compute takes an entire goto_programt, iterates over the instructions and for
+/// any backwards jumps attempts to find out if it's a natural loop.
+///
+/// All instructions in a natural loop are stored into \ref loop_map, keyed by
+/// their header - the target of the backwards goto jump.
 template<class P, class T>
```

Comment:
Might be worth elaborating what P (the program representation) and T (the node type) need to provide for there to be a valid instantiation of this template?